os/linux.go

```diff
@@ -235,7 +235,8 @@ func (c Linux) FileExist(h Host, path string) bool {
 // TODO refactor this into go because it's too magical.
 func (c Linux) LineIntoFile(h Host, path, matcher, newLine string) error {
 	if c.FileExist(h, path) {
-		err := h.Exec(fmt.Sprintf(`/bin/bash -c -- 'file=%s; match=%s; line=%s; grep -q "${match}" "$file" && sed -i "/${match}/c ${line}" -- "$file" || (echo "$line" | tee -a -- "$file" > /dev/null)'`, escape.Quote(path), escape.Quote(matcher), escape.Quote(newLine)))
+		err := h.Exec(fmt.Sprintf(`/bin/bash -c -- 'grep -q "%s" "%s" && sed -i "/%s/c %s" -- "%s" || (echo "%s" | tee -a -- "%s" > /dev/null)'`, 
```

Comment:
Isn't this escaped twice now? Once via escape.Quote() and the second time via the double quotes?

Reply:
I don't think it is. Escaping works as expected for me.